__tests__/app.test.js

```diff
@@ -0,0 +1,44 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed.js");
+const testData = require("../db/data/test-data/index.js");
+
+afterAll(() => {
+  return db.end();
+});
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+describe("Request to non-existent route", () => {
+  test('404: should response with not found when trying to access a non-existent endpoint', () => {
+    return request(app)
+      .get("/apo")
+      .expect(404)
+      .then(({ body: { message } }) => {
+        expect(message).toBe("This page does not exist!");
+      })
+  });
+});
+
+describe("GET /api/topics", () => {
+  test("200: should response with all topics", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body: { topics } }) => {
+        expect(topics).toBeInstanceOf(Array);
```

Comment:
Delete line 32, as we are using a forEach on an array already and we are checking the length of the array.